scripts/firedrake-install

```diff
@@ -588,6 +588,30 @@ def build_and_install_glpsol():
         os.chdir("..")
 
 
+def build_and_install_libspatialindex():
+    log.info("Installing libspatialindex\n")
+    if os.path.exists("libspatialindex"):
+        libspatialindex_changed = git_update("libspatialindex")
+    else:
+        git_clone("git+https://github.com/miklos1/libspatialindex.git")
```

Comment:
We will change this to `github.com/firedrakeproject/libspatialindex` if everything goes well.